packages/calypso-e2e/src/rest-api-client.ts

```diff
@@ -74,13 +77,20 @@ export class RestAPIClient {
 		params.append( 'get_bearer_token', '1' );
 
 		// Request the bearer token for the user.
-		const response: BearerTokenResponse = await this.sendRequest( new URL( BEARER_TOKEN_URL ), {
-			method: 'post',
-			body: params,
-		} );
+		const response: BearerTokenResponse | BearerTokenErrorResponse = await this.sendRequest(
+			new URL( BEARER_TOKEN_URL ),
+			{
+				method: 'post',
+				body: params,
+			}
+		);
+
+		if ( response.success === false ) {
+			const firstError = response.data.errors.at( 0 );
+			throw new Error( `${ firstError?.code }: ${ firstError?.message }` );
+		}
 
-		this.bearerToken = response.data.bearer_token;
-		return this.bearerToken;
+		return response.data.bearer_token;
```

Comment:
So, I changed the signature to only return the bearer token because the `getBearerToken` method's name suggests its purpose, which is to return the bearer token. What the caller function does with the token is not of interest to the `getBearerToken` method.

This change was also made to support this attempt at unit testing the class. Having the side effect of setting the bearer token required network call to the authentication endpoint, which was not desirable from the unit test perspective.